### PR TITLE
feat(build): #1202 update nixpkgs

### PR DESCRIPTION
**Pull request by rohaquinlop**
- Update nixpkgs to last version

### CODE REVIEW

File: src/args/agnostic.nix [hunk 1]
```diff
@@ -10,7 +10,13 @@
   fix' = __unfix__: let x = __unfix__ x // {inherit __unfix__;}; in x;
   sources = import ../nix/sources.nix;
   args = fix' (self: let
-    __nixpkgs__ = import sources.nixpkgs {inherit system;};
+    __nixpkgs__ = import sources.nixpkgs {
+      inherit system;
+      config = {
+        allowUnfree = true;
+        permittedInsecurePackages = ["ruby-2.7.8" "openssl-1.1.1w"];
```
Comment (dsalaza4) [inline src/args/agnostic.nix:17]: Maybe drop ruby 2.7 support?

What is requiring openssl-1.1.1?

Comment (rohaquinlop) [reply src/args/agnostic.nix:17]: We can drop the support because we are adding now ruby 3.1, 3.2 and 3.3

Comment (dsalaza4) [reply src/args/agnostic.nix:17]: Lets only keep officially supported versions https://www.ruby-lang.org/en/downloads/branches/

File: docs/src/api/builtins/utilities.md [hunk 2]
```diff
@@ -59,7 +59,7 @@ m github:fluidattacks/makes@23.07 /utils/makeRubyLock \
   "${sources_yaml}"
 ```
 
-- Supported `ruby_version`s are: `2.7`, `3.0` and `3.1`.
+- Supported `ruby_version`s are: `2.7`, `3.0`, `3.2` and `3.3`.
```
Comment (dsalaza4) [inline docs/src/api/builtins/utilities.md:62]:
```suggestion
- Supported `ruby_version`s are: `3.0`, `3.1`, `3.2` and `3.3`.
```

File: src/args/agnostic.nix (hunk 1 above)
Comment (dsalaza4) [reply src/args/agnostic.nix:17]: Hopefully we can also get rid of that one if we deprecate 2.7

Comment (rohaquinlop) [reply src/args/agnostic.nix:17]: gotcha!

File: docs/src/api/builtins/utilities.md (hunk 2 above)
Comment (rohaquinlop) [reply docs/src/api/builtins/utilities.md]: ruby 3.0 was removed from nixpkgs

File: src/args/agnostic.nix (hunk 1 above)
Comment (rohaquinlop) [reply src/args/agnostic.nix]: removed!

File: src/args/agnostic.nix [hunk 3]
```diff
@@ -10,7 +10,10 @@
   fix' = __unfix__: let x = __unfix__ x // {inherit __unfix__;}; in x;
   sources = import ../nix/sources.nix;
   args = fix' (self: let
-    __nixpkgs__ = import sources.nixpkgs {inherit system;};
+    __nixpkgs__ = import sources.nixpkgs {
+      inherit system;
+      config = {allowUnfree = true;};
```
Comment (dsalaza4) [inline src/args/agnostic.nix:15]:
```suggestion
      config.allowUnfree = true;
```

File: makes/utils/makeNodeJsLock/entrypoint.sh [hunk 4]
```diff
@@ -9,11 +9,11 @@ function main {
   )
 
   : && case "${node_js_version}" in
-      14) npm=__argNode14__/bin/npm ;;
-      16) npm=__argNode16__/bin/npm ;;
-      18) npm=__argNode18__/bin/npm ;;
-      *) critical NodeJs version not supported: "${node_js_version}" ;;
-    esac \
+    18) npm=__argNode18__/bin/npm ;;
```
Comment (dsalaza4) [inline makes/utils/makeNodeJsLock/entrypoint.sh:12]: wrong indentation?

**Review by dsalaza4 — APPROVED** [review]
lgtm

**Pull request merged**